presto-main/src/main/java/io/prestosql/operator/window/LagFunction.java

```diff
@@ -50,9 +52,23 @@ public void processRow(BlockBuilder output, int frameStart, int frameEnd, int cu
             long offset = (offsetChannel < 0) ? 1 : windowIndex.getLong(offsetChannel, currentPosition);
             checkCondition(offset >= 0, INVALID_FUNCTION_ARGUMENT, "Offset must be at least 0");
 
-            long valuePosition = currentPosition - offset;
+            long valuePosition;
 
-            if ((valuePosition >= 0) && (valuePosition <= currentPosition)) {
+            if (ignoreNulls) {
+                long count = 0;
+                valuePosition = currentPosition - 1;
+                while (withinPartition(valuePosition, currentPosition)) {
+                    if (!windowIndex.isNull(valueChannel, toIntExact(valuePosition)) && ++count == offset) {
```

Comment:
Nit: in our code style, we don't use the increment operations inside expressions. They should be separate statements. Same applies to the other functions.